agenta-web/src/lib/helpers/utils.ts

```diff
@@ -313,3 +314,16 @@ export const shortPoll = async (
         await delay(delayMs)
     }
 }
+
+export const generateOrRetrieveDistinctId = (): string => {
+    if (typeof localStorage !== "undefined") {
+        let distinctId = localStorage.getItem("posthog_distinct_id")
```

Comment:
why do we need to generate a posthog distinct id in this case, I though if we leave it empty posthog would take care of all of this automatically

Reply:
@aybruhm reminder

Reply:
I made changes because PostHog wasn't keeping IDs unique for user requests. 

When I tested it without the current code changes, I noticed that after a while, PostHog started using a different ID for the events been triggered. So, I figured it'd be better to generate a UUID for each user, stick it to their browser, and let PostHog use that as their distinct ID. This way, even if a user takes off for a few days and comes back (unless they clear their browser storage), PostHog will keep on tracking them with the generated UUID as their ID.